internal/resolve/catalog.go

```diff
@@ -114,33 +120,47 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 		if len(resolvedBundles) != 0 {
 			// We've already found one or more package candidates
 			currentIsDeprecated := isDeprecated(thisBundle, thisDeprecation)
-			priorIsDeprecated := isDeprecated(*resolvedBundles[len(resolvedBundles)-1], priorDeprecation)
+			priorIsDeprecated := isDeprecated(*resolvedBundles[len(resolvedBundles)-1].bundle, priorDeprecation)
 			if currentIsDeprecated && !priorIsDeprecated {
 				// Skip this deprecated package and retain the non-deprecated package(s)
 				return nil
 			} else if !currentIsDeprecated && priorIsDeprecated {
 				// Our package candidates so far were deprecated and this one is not; clear the lists
-				resolvedBundles = []*declcfg.Bundle{}
-				matchedCatalogs = []string{}
+				resolvedBundles = []foundBundle{}
 			}
 		}
 		// The current bundle shares deprecation status with prior bundles or
 		// there are no prior bundles. Add it to the list.
-		resolvedBundles = append(resolvedBundles, &thisBundle)
-		matchedCatalogs = append(matchedCatalogs, cat.GetName())
+		resolvedBundles = append(resolvedBundles, foundBundle{&thisBundle, cat.GetName(), cat.Spec.Priority})
 		priorDeprecation = thisDeprecation
 		return nil
 	}, listOptions...); err != nil {
 		return nil, nil, nil, fmt.Errorf("error walking catalogs: %w", err)
 	}
 
+	// Resolve for priority
+	if len(resolvedBundles) > 1 {
```

Comment:
Couldn't we resolve for priority while we're in the process of walking through the catalogs? Similar to how we ignore a bundle when it's deprecated and the last one was not, couldn't we ignore a bundle when the priority is lower than the last resolved bundle (and it shares deprecation status)? Not that it gains us much because we don't have that many catalogs to sort through later, but just a thought.

Reply:
We still want to be able to list the catalogs, including those that don't have a maximal priority. If you see the test I added, `TestMultiplePriority`, all 3 catalogs are listed, not just the two highest priority. If this is not what we want, then, yes, it might make sense to adjust the list as we create it. Keeping it separate like this, makes it easier to follow, IMHO.

Reply:
Tradeoff between performance and simplicity/readability.